components/script/dom/gpudevice.rs

```diff
@@ -214,6 +220,139 @@ impl GPUDevice {
         }
     }
 
+    fn parse_render_pipeline<'c>(
+        &self,
+        descriptor: &GPURenderPipelineDescriptor,
+    ) -> (
+        Option<(PipelineLayoutId, Vec<BindGroupLayoutId>)>,
+        Option<RenderPipelineDescriptor<'c>>,
+    ) {
+        let mut valid = true;
+
+        let (layout, implicit_ids, _) = self.get_pipeline_layout_data(&descriptor.parent.layout);
+
+        let desc = if valid {
```

Comment:
Valid is always true here. What is the intention behind this code?

Reply:
Ah, I planned to use https://docs.rs/wgpu/latest/wgpu/struct.ColorWrites.html#method.from_bits_retain to use validness check in wgpu-core, but I forgot.

Reply:
And it makes `webgpu:api,validation,render_pipeline,fragment_state:targets_write_mask:*` work.